src/engines/engine_models.rs

```diff
@@ -15,73 +15,24 @@ use scraper::error::SelectorErrorKind;
 /// and are errors mostly related to failure in initialization of HeaderMap, Selector errors and
 /// all other errors occuring within the code handling the `upstream search engines`.
 #[derive(Debug)]
-pub enum EngineErrorKind {
-    RequestError(reqwest::Error),
+pub enum EngineError {
     EmptyResultSet,
-    UnexpectedError {
-        message: String,
-        source: Option<Box<dyn std::error::Error>>,
-    },
+    RequestError,
+    UnexpectedError,
 }
 
-/// Implementing `Display` trait to make errors writable on the stdout and also providing/passing the
-/// appropriate errors that should be written to the stdout when this error is raised/encountered.
-impl std::fmt::Display for EngineErrorKind {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+impl fmt::Display for EngineError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            EngineErrorKind::RequestError(request_error) => {
-                write!(f, "Request error: {}", request_error)
-            }
-            EngineErrorKind::EmptyResultSet => {
+            EngineError::EmptyResultSet => {
                 write!(f, "The upstream search engine returned an empty result set")
             }
-            EngineErrorKind::UnexpectedError { message, source } => {
-                write!(f, "Unexpected error: {}", message)?;
-                if let Some(source) = source {
-                    write!(f, "\nCaused by: {}", source)?;
-                }
-                Ok(())
+            EngineError::RequestError => {
+                write!(f, "Request error")
             }
+            EngineError::UnexpectedError => write!(f, "Unexpected error"),
```

Comment:
```suggestion
            EngineError::RequestError => {
                write!(f, "Error occurred while requesting data from upstream search engine")
            }
            EngineError::UnexpectedError => write!(f, "An unexpected error occurred while processing the data"),
```

I think we can make the errors a little bit more descriptive like not adding too much detail but enough for the user to understand as well and this will also allow us to better understand when somebody opens a bug report when any one of these issue occurs :slightly_smiling_face:, What do you say ??